### ISSUE TITLE
Violin plots changed as of Seaborn 0.13. [Tests broken].

### ISSUE THREAD

**Issue by SylviaWhittle** [AFM-SPM/TopoStats#679]
In making PR #678 , I believe I have discovered a bug in the testing suite & local testing if the Seaborn package is updated.

I was messing about with plotting as I had found a bug caused by an exception thrown by Seaborn regarding KDE plots, and I realised I had not updated Seaborn to check if the issue was resolved.

After updating Seaborn, I found there were a new couple of tests that failed, namely:
- `tests/test_plotting.py::test_plot_violin`
- `tests/test_plotting.py::test_plot_violin_multiple_directories`

The versions of Seaborn in question are: `0.12` and `0.13`.

The differences in the violin plots are:
Multiple:
Old:
![image](https://github.com/AFM-SPM/TopoStats/assets/86117496/65ac2cb4-b86d-4244-b3ab-aa4164920c27)
New:
![image](https://github.com/AFM-SPM/TopoStats/assets/86117496/10a2acd6-0f76-4ceb-8e24-ca576591a07f)

Single:
Old:
![image](https://github.com/AFM-SPM/TopoStats/assets/86117496/6ab2c819-341c-42d4-9a48-4ab945eba215)
New:
![image](https://github.com/AFM-SPM/TopoStats/assets/86117496/c8dd5262-008e-4b84-8d4c-7c951861eb5a)

I haven't yet checked online to see if others are experiencing this issue or if `0.13` is a breaking release.

We could maybe fix this by just manually adding a legend in the code and updating the test plots. I will try this tomorrow, as I have been looking into this too long already.

Alternatively we could fix the version for Seaborn at `0.12` and descend further into dependancy hell.

The test suite in GitHub Actions fails too, as you can see in the presenting symptom: https://github.com/AFM-SPM/TopoStats/actions/runs/6399786751/job/17372413989?pr=678. Which would need fixing in version too if we went that route.

Perhaps I'm just being silly and it's a trivial matter though!

Thoughts, @ns-rse ?

Branch: Main, Current, 2.1.2.dev41+gda58094ab
OS: MacOS, Ubuntu, Windows
Config: Default
Python Versions: 3.8, 3.9, 3.10, 3.11 (Courtesy of GH-A) 

Here is a clean log of evidence for the issue on unaltered `main`:
```bash
    ~/Documents/TopoStats    main *23 ?1                                                                                                                                                                                                        new   01:50:37  
❯ pip install seaborn==0.12  
Collecting seaborn==0.12
  Downloading seaborn-0.12.0-py3-none-any.whl (285 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 285.1/285.1 kB 2.4 MB/s eta 0:00:00
Requirement already satisfied: numpy>=1.17 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn==0.12) (1.23.4)
Requirement already satisfied: pandas>=0.25 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn==0.12) (1.5.3)
Requirement already satisfied: matplotlib>=3.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn==0.12) (3.7.1)
Requirement already satisfied: contourpy>=1.0.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (1.0.7)
Requirement already satisfied: cycler>=0.10 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (4.39.2)
Requirement already satisfied: kiwisolver>=1.0.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (1.4.4)
Requirement already satisfied: packaging>=20.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (23.0)
Requirement already satisfied: pillow>=6.2.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (9.4.0)
Requirement already satisfied: pyparsing>=2.3.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib>=3.1->seaborn==0.12) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from pandas>=0.25->seaborn==0.12) (2022.7.1)
Requirement already satisfied: six>=1.5 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from python-dateutil>=2.7->matplotlib>=3.1->seaborn==0.12) (1.16.0)
Installing collected packages: seaborn
  Attempting uninstall: seaborn
    Found existing installation: seaborn 0.13.0
    Uninstalling seaborn-0.13.0:
      Successfully uninstalled seaborn-0.13.0
Successfully installed seaborn-0.12.0
    ~/Documents/TopoStats    main *23 ?1                                                                                                                                                                                                        new   01:50:49  
❯ pytest tests/test_plotting.py
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.9, pytest-7.2.2, pluggy-1.0.0
Matplotlib: 3.7.1
Freetype: 2.6.1
rootdir: /Users/sylvi/Documents/TopoStats, configfile: pyproject.toml
plugins: regtest-1.5.0, xdist-3.2.1, mpl-0.16.1, durations-1.2.0, cov-4.0.0, anyio-3.6.2, lazy-fixture-0.6.3
collected 20 items                                                                                                                                                                                                                                                           

tests/test_plotting.py ....................                                                                                                                                                                                                                            [100%]

============================================================================================================================ fixture duration top ============================================================================================================================
total          name                                    num avg            min           
0:00:00.103808                          summary_config  17 0:00:00.005911 0:00:00.005544
0:00:00.057294     toposum_object_multiple_directories  12 0:00:00.004351 0:00:00.003898
0:00:00.015741         toposum_object_single_directory   4 0:00:00.003893 0:00:00.003613
0:00:00.178921                             grand total  43 0:00:00.001815 0:00:00.000020
=========================================================================================================================== test call duration top ===========================================================================================================================
total          name                                    num avg            min           
0:00:00.639732                            test_toposum   1 0:00:00.639732 0:00:00.639732
0:00:00.224771 test_plot_hist_kde_multiple_directories   1 0:00:00.224771 0:00:00.224771
0:00:00.216870                           test_plot_kde   1 0:00:00.216870 0:00:00.216870
0:00:00.199390   test_plot_violin_multiple_directories   1 0:00:00.199390 0:00:00.199390
0:00:00.199074      test_plot_kde_multiple_directories   1 0:00:00.199074 0:00:00.199074
0:00:00.193684     test_plot_hist_multiple_directories   1 0:00:00.193684 0:00:00.193684
0:00:00.167039                        test_plot_violin   1 0:00:00.167039 0:00:00.167039
0:00:00.165812                      test_plot_hist_kde   1 0:00:00.165812 0:00:00.165812
0:00:00.153714                          test_plot_hist   1 0:00:00.153714 0:00:00.153714
0:00:00.032053                test_var_to_label_config   1 0:00:00.032053 0:00:00.032053
0:00:00.009629                          test_melt_data   1 0:00:00.009629 0:00:00.009629
0:00:02.206391                             grand total  20 0:00:00.153714 0:00:00.000063
========================================================================================================================== test setup duration top ===========================================================================================================================
total          name                                    num avg            min           
0:00:00.044048                          test_set_label   4 0:00:00.010957 0:00:00.010605
0:00:00.019268                        test_df_from_csv   1 0:00:00.019268 0:00:00.019268
0:00:00.010958                      test_toposum_class   1 0:00:00.010958 0:00:00.010958
0:00:00.010723      test_plot_kde_multiple_directories   1 0:00:00.010723 0:00:00.010723
0:00:00.010533                            test_outfile   1 0:00:00.010533 0:00:00.010533
0:00:00.010486                           test_plot_kde   1 0:00:00.010486 0:00:00.010486
0:00:00.010481                 test_set_label_keyerror   1 0:00:00.010481 0:00:00.010481
0:00:00.010354 test_plot_hist_kde_multiple_directories   1 0:00:00.010354 0:00:00.010354
0:00:00.009834                        test_plot_violin   1 0:00:00.009834 0:00:00.009834
0:00:00.009820   test_plot_violin_multiple_directories   1 0:00:00.009820 0:00:00.009820
0:00:00.009779     test_plot_hist_multiple_directories   1 0:00:00.009779 0:00:00.009779
0:00:00.009568                          test_plot_hist   1 0:00:00.009568 0:00:00.009568
0:00:00.009343                      test_plot_hist_kde   1 0:00:00.009343 0:00:00.009343
0:00:00.006020                            test_toposum   1 0:00:00.006020 0:00:00.006020
0:00:00.183533                             grand total  20 0:00:00.009834 0:00:00.000068
========================================================================================================================= test teardown duration top =========================================================================================================================
total          name                                    num avg            min           
0:00:00.002600                             grand total  20 0:00:00.000151 0:00:00.000046

---------- coverage: platform darwin, python 3.10.9-final-0 ----------
Name                                Stmts   Miss  Cover
-------------------------------------------------------
topostats/__main__.py                   2      2     0%
topostats/entry_point.py               79     38    52%
topostats/filters.py                  153    132    14%
topostats/grains.py                   105     81    23%
topostats/grainstats.py               331    281    15%
topostats/io.py                       386    299    23%
topostats/logs/logs.py                 25      0   100%
topostats/plotting.py                 144     13    91%
topostats/plottingfuncs.py            115     95    17%
topostats/processing.py               213    189    11%
topostats/run_topostats.py            113     94    17%
topostats/scars.py                    105     94    10%
topostats/statistics.py                26     20    23%
topostats/theme.py                     40     23    42%
topostats/thresholds.py                31     18    42%
topostats/tracing/dnacurvature.py      24     24     0%
topostats/tracing/dnatracing.py       413    364    12%
topostats/tracing/skeletonize.py       23     12    48%
topostats/tracing/tracingfuncs.py     613    550    10%
topostats/utils.py                     76     44    42%
topostats/validation.py                15      5    67%
-------------------------------------------------------
TOTAL                                3032   2378    22%

============================================================================================================================= 20 passed in 4.85s =============================================================================================================================
    ~/Documents/TopoStats    main *23 ?1                                                                                                                                                                                                 7s   new   01:51:00  
❯ pip install --upgrade seaborn
Requirement already satisfied: seaborn in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (0.12.0)
Collecting seaborn
  Obtaining dependency information for seaborn from https://files.pythonhosted.org/packages/7b/e5/83fcd7e9db036c179e0352bfcd20f81d728197a16f883e7b90307a88e65e/seaborn-0.13.0-py3-none-any.whl.metadata
  Using cached seaborn-0.13.0-py3-none-any.whl.metadata (5.3 kB)
Requirement already satisfied: numpy!=1.24.0,>=1.20 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn) (1.23.4)
Requirement already satisfied: pandas>=1.2 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn) (1.5.3)
Requirement already satisfied: matplotlib!=3.6.1,>=3.3 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from seaborn) (3.7.1)
Requirement already satisfied: contourpy>=1.0.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (1.0.7)
Requirement already satisfied: cycler>=0.10 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (4.39.2)
Requirement already satisfied: kiwisolver>=1.0.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (1.4.4)
Requirement already satisfied: packaging>=20.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (23.0)
Requirement already satisfied: pillow>=6.2.0 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (9.4.0)
Requirement already satisfied: pyparsing>=2.3.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from matplotlib!=3.6.1,>=3.3->seaborn) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from pandas>=1.2->seaborn) (2022.7.1)
Requirement already satisfied: six>=1.5 in /opt/homebrew/Caskroom/miniforge/base/envs/new/lib/python3.10/site-packages (from python-dateutil>=2.7->matplotlib!=3.6.1,>=3.3->seaborn) (1.16.0)
Using cached seaborn-0.13.0-py3-none-any.whl (294 kB)
Installing collected packages: seaborn
  Attempting uninstall: seaborn
    Found existing installation: seaborn 0.12.0
    Uninstalling seaborn-0.12.0:
      Successfully uninstalled seaborn-0.12.0
Successfully installed seaborn-0.13.0
    ~/Documents/TopoStats    main *23 ?1                                                                                                                                                                                                        new   01:51:14  
❯ pytest tests/test_plotting.py
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.9, pytest-7.2.2, pluggy-1.0.0
Matplotlib: 3.7.1
Freetype: 2.6.1
rootdir: /Users/sylvi/Documents/TopoStats, configfile: pyproject.toml
plugins: regtest-1.5.0, xdist-3.2.1, mpl-0.16.1, durations-1.2.0, cov-4.0.0, anyio-3.6.2, lazy-fixture-0.6.3
collected 20 items                                                                                                                                                                                                                                                           

tests/test_plotting.py ..................FF                                                                                                                                                                                                                            [100%]

================================================================================================================================== FAILURES ==================================================================================================================================
______________________________________________________________________________________________________________________________ test_plot_violin ______________________________________________________________________________________________________________________________
Error: Image files did not match.
  RMS Value: 54.82801895050593
  Expected:  
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin/baseline.png
  Actual:    
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin/result.png
  Difference:
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin/result-failed-diff.png
  Tolerance: 
    2
--------------------------------------------------------------------------------------------------------------------------- Captured stdout setup ----------------------------------------------------------------------------------------------------------------------------
[Wed, 04 Oct 2023 01:51:23] [INFO    ] [topostats] [plotting] Data has been melted to long format for plotting.
[Wed, 04 Oct 2023 01:51:23] [INFO    ] [topostats] [plotting] Seaborn color palette : bright
--------------------------------------------------------------------------------------------------------------------------- Captured stderr setup ----------------------------------------------------------------------------------------------------------------------------
INFO:topostats:[plotting] Data has been melted to long format for plotting.
INFO:topostats:[plotting] Seaborn color palette : bright
----------------------------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------------------------
INFO     topostats:plotting.py:194 [plotting] Data has been melted to long format for plotting.
INFO     topostats:plotting.py:216 [plotting] Seaborn color palette : bright
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
[Wed, 04 Oct 2023 01:51:23] [INFO    ] [topostats] [plotting] Plotted area to : output/summary_distributions/area_violin.png
---------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------
INFO:topostats:[plotting] Plotted area to : output/summary_distributions/area_violin.png
___________________________________________________________________________________________________________________ test_plot_violin_multiple_directories ____________________________________________________________________________________________________________________
Error: Image files did not match.
  RMS Value: 45.48112799099571
  Expected:  
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin_multiple_directories/baseline.png
  Actual:    
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin_multiple_directories/result.png
  Difference:
    /var/folders/sr/wjtfqr9s6x3bw1s647t649x80000gn/T/tmpgeo326in/test_plotting.test_plot_violin_multiple_directories/result-failed-diff.png
  Tolerance: 
    2
--------------------------------------------------------------------------------------------------------------------------- Captured stdout setup ----------------------------------------------------------------------------------------------------------------------------
[Wed, 04 Oct 2023 01:51:23] [INFO    ] [topostats] [plotting] Data has been melted to long format for plotting.
[Wed, 04 Oct 2023 01:51:23] [INFO    ] [topostats] [plotting] Seaborn color palette : bright
--------------------------------------------------------------------------------------------------------------------------- Captured stderr setup ----------------------------------------------------------------------------------------------------------------------------
INFO:topostats:[plotting] Data has been melted to long format for plotting.
INFO:topostats:[plotting] Seaborn color palette : bright
----------------------------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------------------------
INFO     topostats:plotting.py:194 [plotting] Data has been melted to long format for plotting.
INFO     topostats:plotting.py:216 [plotting] Seaborn color palette : bright
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
[Wed, 04 Oct 2023 01:51:24] [INFO    ] [topostats] [plotting] Plotted area to : output/summary_distributions/area_violin.png
---------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------
INFO:topostats:[plotting] Plotted area to : output/summary_distributions/area_violin.png
============================================================================================================================ fixture duration top ============================================================================================================================
total          name                                    num avg            min           
0:00:00.099537                          summary_config  17 0:00:00.005724 0:00:00.005486
0:00:00.050189     toposum_object_multiple_directories  12 0:00:00.004106 0:00:00.003654
0:00:00.015708         toposum_object_single_directory   4 0:00:00.003982 0:00:00.003608
0:00:00.166808                             grand total  43 0:00:00.001144 0:00:00.000018
=========================================================================================================================== test call duration top ===========================================================================================================================
total          name                                    num avg            min           
0:00:00.580012                            test_toposum   1 0:00:00.580012 0:00:00.580012
0:00:00.223636                           test_plot_kde   1 0:00:00.223636 0:00:00.223636
0:00:00.222772 test_plot_hist_kde_multiple_directories   1 0:00:00.222772 0:00:00.222772
0:00:00.217245      test_plot_kde_multiple_directories   1 0:00:00.217245 0:00:00.217245
0:00:00.208911     test_plot_hist_multiple_directories   1 0:00:00.208911 0:00:00.208911
0:00:00.192318   test_plot_violin_multiple_directories   1 0:00:00.192318 0:00:00.192318
0:00:00.176896                      test_plot_hist_kde   1 0:00:00.176896 0:00:00.176896
0:00:00.167456                          test_plot_hist   1 0:00:00.167456 0:00:00.167456
0:00:00.165274                        test_plot_violin   1 0:00:00.165274 0:00:00.165274
0:00:00.027782                test_var_to_label_config   1 0:00:00.027782 0:00:00.027782
0:00:02.189937                             grand total  20 0:00:00.165274 0:00:00.000074
========================================================================================================================== test setup duration top ===========================================================================================================================
total          name                                    num avg            min           
0:00:00.040354                          test_set_label   4 0:00:00.010101 0:00:00.009847
0:00:00.012039      test_plot_kde_multiple_directories   1 0:00:00.012039 0:00:00.012039
0:00:00.011544                        test_df_from_csv   1 0:00:00.011544 0:00:00.011544
0:00:00.010483                      test_toposum_class   1 0:00:00.010483 0:00:00.010483
0:00:00.010407     test_plot_hist_multiple_directories   1 0:00:00.010407 0:00:00.010407
0:00:00.010335                          test_plot_hist   1 0:00:00.010335 0:00:00.010335
0:00:00.010216   test_plot_violin_multiple_directories   1 0:00:00.010216 0:00:00.010216
0:00:00.010209                      test_plot_hist_kde   1 0:00:00.010209 0:00:00.010209
0:00:00.009986                           test_plot_kde   1 0:00:00.009986 0:00:00.009986
0:00:00.009726                            test_outfile   1 0:00:00.009726 0:00:00.009726
0:00:00.009711 test_plot_hist_kde_multiple_directories   1 0:00:00.009711 0:00:00.009711
0:00:00.009522                 test_set_label_keyerror   1 0:00:00.009522 0:00:00.009522
0:00:00.009284                        test_plot_violin   1 0:00:00.009284 0:00:00.009284
0:00:00.005854                            test_toposum   1 0:00:00.005854 0:00:00.005854
0:00:00.171257                             grand total  20 0:00:00.009986 0:00:00.000080
========================================================================================================================= test teardown duration top =========================================================================================================================
total          name                                    num avg            min           
0:00:00.002588                             grand total  20 0:00:00.000140 0:00:00.000052

---------- coverage: platform darwin, python 3.10.9-final-0 ----------
Name                                Stmts   Miss  Cover
-------------------------------------------------------
topostats/__main__.py                   2      2     0%
topostats/entry_point.py               79     38    52%
topostats/filters.py                  153    132    14%
topostats/grains.py                   105     81    23%
topostats/grainstats.py               331    281    15%
topostats/io.py                       386    299    23%
topostats/logs/logs.py                 25      0   100%
topostats/plotting.py                 144     13    91%
topostats/plottingfuncs.py            115     95    17%
topostats/processing.py               213    189    11%
topostats/run_topostats.py            113     94    17%
topostats/scars.py                    105     94    10%
topostats/statistics.py                26     20    23%
topostats/theme.py                     40     23    42%
topostats/thresholds.py                31     18    42%
topostats/tracing/dnacurvature.py      24     24     0%
topostats/tracing/dnatracing.py       413    364    12%
topostats/tracing/skeletonize.py       23     12    48%
topostats/tracing/tracingfuncs.py     613    550    10%
topostats/utils.py                     76     44    42%
topostats/validation.py                15      5    67%
-------------------------------------------------------
TOTAL                                3032   2378    22%

========================================================================================================================== short test summary info ===========================================================================================================================
FAILED tests/test_plotting.py::test_plot_violin
FAILED tests/test_plotting.py::test_plot_violin_multiple_directories
======================================================================================================================== 2 failed, 18 passed in 5.24s ========================================================================================================================
    ~/Documents/TopoStats    main *23 ?1                                                                                                                                                                                                 7s   new   01:51:24  
❯ 
```

**Comment by ns-rse** [comment]
Noticed this on the tests run on `pre-commit-autoupdate-ci`too.

Looks like a change in the defaults for Seaborn and a correction/improvement to multiple plotting (in our test image the violins don't align with the marks on the axes).

I'm happy to take this (need something simple to ease back into coding) and think updating our test images and setting a default to include a legend in multiple violin plots would be sensible.

Once merged the pre-commit and your #678 can be updated to `main` and they should then pass the tests and we can make a bug fix release.

**Comment by SylviaWhittle** [comment]
Thanks @ns-rse !
